app/models/automated_checks/dqt_report_consumer.rb

```diff
@@ -19,17 +19,16 @@ def ingest
       return if errors.any?
 
       claims = Claim.awaiting_task("qualifications")
-      csv.rows.each do |row|
-        reference = row.fetch("dfeta text2")
-        qts_date = row.fetch("dfeta qtsdate")
-        claim = claims.detect { |c| c.reference == reference }
-        next if qts_date.blank? || claim.nil?
-        if claim.policy::DQTRecord.new(row.to_h).eligible? && row_matches_claim?(row, claim)
+      records = DQTReportCsvToRecords.new(csv.rows).transform
+      records.each do |record|
+        claim = claims.detect { |c| c.reference == record.fetch(:claim_reference) }
+        next if record.fetch(:qts_date).blank? || claim.nil?
```

Comment:
Should we add `next if record.fetch(:qts_date).blank?` to top of the loop? If we're not going to do anything with this record, there's no point trying to find the claim.

Reply:
I wanted to avoid too many `next if` conditionals. I don't think it's too computationally intensive to find the claim in an array of objects, the query fetching all the claims happens outside the loop now. I'll do some quick research to see if this should be optimised though

Reply:
👍